### ISSUE TITLE
Поиск адреса если известен город

### ISSUE THREAD

**Issue by mastaru** [fias-api/jquery#11]
Здравствуйте, не могу понять как реализовать такую вещь, у нас на сайте стоит автоопределение города, при входе на сайт определяется город, например Ростов-на-Дону, как сделать что бы показывались улицы только Ростова-на-Дону, без введение города в input и указания кода города?

**Comment by blackseabreathe** [comment]
получить  кладр и окато код города

`// fias
//тут город краснодар и его кладр код - 2300000100000 и его окато код - 03401000000
if($('input[name="address"]').length>0){
var $address = $('input[name="address"]');
$address.fias({
oneString: true,
parentType: $.fias.type.city, // необходимо
parentId: '2300000100000', // необходимо
okatoId: '03401000000', // необходимо
});
}`

коды городов найдете в интернете

**Comment by mastaru** [comment]
А без кодов города нельзя? Я же когда начинаю вводить в поле название города как-то же коды получаются автоматически... Нельзя сделать запрос сразу и получить улицы этого города, без ввода названия города?

**Comment by blackseabreathe** [comment]
Нет, так как поиск адресов осуществляться должен только по одному городу. Тем не менее если есть опция, которая сразу ограничивает поиск по коду города, почему ей не воспользоваться?

И это ничего не поменяет. Ошибка api все равно появляется. Пробовал

**Comment by mastaru** [comment]
Ну потому что этих городов ОЧЕНЬ много, нужны ВСЕ города России и крупные и мелкие из области, мне теперь надо вручную ВСЕ города прописывать с кодами? Ну у Вас же как то по событию изменения в input запросы отправляются же. Затем из списка я выбираю город и код города автоматически подставляется для адреса. Можно немного допилить скрипт что бы запрос на город отправлялся не по событию изменения в input, а по текстовому значению в переменной?

**Comment by blackseabreathe** [comment]
Извините, я не совсем понимаю при чем тут выборка по городу если основная проблема заключается в ошибке api.

**Comment by mastaru** [comment]
Да ошибки нет, я хочу что бы разработчики немного допили скрипт, что бы можно было в функцию fias передать только названия города, без кода и тогда появилась выборка улиц по этому городу...

**Comment by blackseabreathe** [comment]
Странно. Сейчас нет ошибки в консоле.

«что бы разработчики немного допили скрипт, что бы можно было в функцию fias передать только названия города, без кода» 

То есть писать название города кириллицей/латиницей и чтобы код подгружался (допустим из базы) автоматически?

Типа так?

$address.fias({
parentId: ‘Krasnodar’,
});


И где-то на стороне api

parentId = parentId.getValue();

Далее поиск по базе

If (parentId.length){
newParentId = ‘найденный код'
}

И затем

parentId = newParentId;

**Comment by mastaru** [comment]
Ну типа такого, но только по Русски. Я могу в input в какой-нибудь аттрибут передать название города если надо. У нас на сайте автоматом город опредиляется, или же его можно выбрать из списка. А в корзине надо повторно вводить название города... Хотелось бы что бы я мог взять название города и передать получить все его улицы

**Comment by blackseabreathe** [comment]
Так если у Вас где-то на странице корзины уже есть название города, то берите его оттуда и вставляйте в поле. Или я как-то не так Вас понял

**Comment by mastaru** [comment]
Ну так я вставляю через JS но оно не передаётся, там видимо через событие onchange передаётся название города

**Comment by blackseabreathe** [comment]
Если покажите кусок исходного кода и ссылку на сайт, может смогу понять почему не подтягивается город.

**Comment by 1234ru** [comment]
Похоже, у меня была такая же проблема, как у автора темы. Попробую изложить по-своему.

Если поле при загрузке страницы уже заполнено, библиотека его автоматически не подхватывает и не использует при подборе вариантов для дочерних полей, в результате чего автозаполнение для них не работает.

Например, если заполнен город, то нельзя просто взять и начать заполнять улицу. Нужно вернуться к городу, стереть его, начать заполнять заново, выбрать значение из списка для подстановки. Затем перейти к улице, и вот только тогда для улицы тоже возникнет список для подстановки.

Получается, что уже заполненное поле оказыавется менее удобным, чем не заполненное. Ситуация абсурдная.

Чтобы этот вопрос решить, нужно вручную запрашивать данные для каждого из полей с помощью метода `$.fias.api()`. Вот пример:

```html
<form>
    <input type="text" name="city" value="Москва">
    <input type="text" name="street">
</form>
```

```javascript
// Скрипт IE11 friendly, поэтому синтаксис старый
var $container = $('form'),
    $city = $container.find('[name="city"]'),
    $street = $container.find('[name="street"]');

$().add($city).add($street);

$city.fias('type', $.fias.type.city);
$street.fias('type', $.fias.type.street);
$street.fias('parentType', $.fias.type.city);

var query = {
    'type': 'city',
    'name': $city.val(),
    'limit': 1, // при точном совпадении достаточно одного элемента
};

$.fias.api(query, function (response) {
    // В бесплатной версии первым пунктом всегда приходит
    // "Бесплатная версия...", независимо от limit,
    // поэтому берем последний элемент.
    var item = response[ response.length - 1];

    // Проверяем совпадение
    if (item && item.name == query.name) {
        $street.fias('parentId', item.id);
        // Вот теперь подстановка улицы будет работать
    }
});

```

**Comment by 1234ru** [comment]
Пример из [комментария выше](https://github.com/fias-api/jquery/issues/11#issuecomment-1247992199) выше изолированный.

Чтобы было удобней этим пользоваться на практике, я оформил свои разработки в виде небольшой библиотечки: https://gist.github.com/1234ru/85d64980510ac8a3bb370272fdf00a81

Пользоваться примерно так:

```javascript
var fieldsHierarchy = [
    {
        type: 'city',
        selector: 'input[name="city"]',
        properties: { 'withParents': true },
    },
    {
        type: 'street',
        // Вместо selector передаём поле в явном виде:
        fieldObject: $('input[name="street"]'),
    },
    {
        type: 'building',
        // если не указать ни selector, ни fieldObject
        // будет использовано input[name={type}]
    },
];

var fiasSettings = {
    change: function (obj) {
        console.log("Some FIAS field has changed")
    }
};

var obj = new _Local.FIAS().setAutoComplete(
    fieldsHierarchy,
    fiasSettings
);
// Вуаля! У вас подстановка адресов с учетом
// заполненных на момент загрузки страницы полей.
```

Код написан в старом синтаксисе для поддержки IE11 без использования компиляторов.
